messaging/src/main/java/org/axonframework/eventhandling/processors/streaming/pooled/SplitTask.java

```diff
@@ -120,14 +120,19 @@ private boolean splitAndRelease(Segment segmentToSplit) {
                 unitOfWorkFactory
                         .create()
                         .executeWithResult(context -> {
-                            TrackingToken tokenToSplit = tokenStore.fetchToken(name, segmentToSplit.getSegmentId());
+                            TrackingToken tokenToSplit = joinAndUnwrap(
+                                    tokenStore.fetchToken(name, segmentToSplit.getSegmentId(), null)
+                            );
                             TrackerStatus[] splitStatuses = TrackerStatus.split(segmentToSplit, tokenToSplit);
-                            tokenStore.initializeSegment(
+                            joinAndUnwrap(tokenStore.initializeSegment(
                                     splitStatuses[1].getTrackingToken(),
                                     name,
-                                    splitStatuses[1].getSegment().getSegmentId()
-                            );
-                            tokenStore.releaseClaim(name, splitStatuses[0].getSegment().getSegmentId());
+                                    splitStatuses[1].getSegment().getSegmentId(),
+                                    null
+                            ));
+                            joinAndUnwrap(tokenStore.releaseClaim(name,
+                                                                  splitStatuses[0].getSegment().getSegmentId(),
+                                                                  null));
```

Comment:
```suggestion
                                                                  context));
```

We should pass the `ProcessingContext` here.